Jenkinsfile

```diff
@@ -242,19 +242,21 @@ def runBuildAndTest(Map args = [:]) {
   def filterStage = args.get('filterStage', 'mandatory')
   deleteDir()
   unstashV2(name: 'source', bucket: "${JOB_GCS_BUCKET}", credentialsId: "${JOB_GCS_CREDENTIALS}")
-  dir("${BASE_DIR}"){
-    def mapParallelTasks = [:]
-    def content = readYaml(file: 'Jenkinsfile.yml')
-    if (content?.disabled?.when?.labels && beatsWhen(project: 'top-level', content: content?.disabled?.when)) {
-      error 'Pull Request has been configured to be disabled when there is a skip-ci label match'
-    } else {
-      content['projects'].each { projectName ->
-        generateStages(project: projectName, changeset: content['changeset'], filterStage: filterStage).each { k,v ->
-          mapParallelTasks["${k}"] = v
+  withGCPEnv(secret: 'secret/observability-team/ci/elastic-observability-account-auth'){
```

Comment:
`runBuildAndTest` is a meta-function in charge of reading `Jenkinsfile.yml` files and transform them in pipeline stages.

`withGCPEnv` cannot be used within this context, since every pipeline stage uses an ephemeral worker, therefore `withGCPEnv` is not within the worker context but outside.

In addition, if `withGCPEnv` is added by default then it will increase the build time for each stage regardless those stages interact with GCP or not.

I'd recommend to use a different approach instead.